client/core/core.go

```diff
@@ -436,16 +441,22 @@ func (dc *dexConnection) runMatches(tradeMatches map[order.OrderID]*serverMatche
 		if sm.cancel != nil {
 			err := tracker.processCancelMatch(sm.cancel)
 			if err != nil {
-				return updatedAssets, fmt.Errorf("processCancelMatch for cancel order %v targeting order %v failed: %v",
-					sm.cancel.OrderID, oid, err)
+				return &runMatchResult{
+					updatedAssets: updatedAssets,
+					err: fmt.Errorf("processCancelMatch for cancel order %v targeting order %v failed: %v",
+						sm.cancel.OrderID, oid, err),
+				}
```

Comment:
Thanks for removing the separate receiver goroutines below, but the `runMatches` return really didn't need to change.  This is fine now though.

Reply:
Argh. Just seeing this. Will correct.